product/src/main/java/com/yas/product/exception/InternalServerErrorException.java

```diff
@@ -0,0 +1,17 @@
+package com.yas.product.exception;
+
+import com.yas.product.utils.MessagesUtils;
+
+public class InternalServerErrorException extends RuntimeException {
```

Comment:
`InternalServerErrorException` is generic. I would create a custom exception for specific errors only.
In case of internal server errors, you can just simply throw RuntimeException